seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/source/MongodbSource.java

```diff
@@ -0,0 +1,120 @@
+import org.apache.seatunnel.common.config.CheckResult;
+import org.apache.seatunnel.common.config.TypesafeConfigUtils;
+import org.apache.seatunnel.common.constants.JobMode;
+import org.apache.seatunnel.common.constants.PluginType;
+import org.apache.seatunnel.connectors.seatunnel.common.schema.SeaTunnelSchema;
+import org.apache.seatunnel.connectors.seatunnel.common.source.AbstractSingleSplitReader;
+import org.apache.seatunnel.connectors.seatunnel.common.source.AbstractSingleSplitSource;
+import org.apache.seatunnel.connectors.seatunnel.common.source.SingleSplitReaderContext;
+import org.apache.seatunnel.connectors.seatunnel.mongodb.config.MongodbParameters;
+import org.apache.seatunnel.format.json.JsonDeserializationSchema;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+import org.apache.seatunnel.shade.com.typesafe.config.ConfigBeanFactory;
+
+import com.google.auto.service.AutoService;
+
+@AutoService(SeaTunnelSource.class)
+public class MongodbSource extends AbstractSingleSplitSource<SeaTunnelRow> {
+
+    private SeaTunnelContext seaTunnelContext;
+
+    private SeaTunnelRowType rowType;
+
+    private MongodbParameters params;
+
+    private DeserializationSchema<SeaTunnelRow> deserializationSchema;
+
+    private static final String CONF_PREFIX = "readconfig.";
+
+    @Override
+    public String getPluginName() {
+        return "MongoDB";
+    }
+
+    @Override
+    public void prepare(Config config) throws PrepareFailException {
+        Config extractConfig = TypesafeConfigUtils.extractSubConfig(config, CONF_PREFIX, false);
+
+        CheckResult result = CheckConfigUtil.checkAllExists(extractConfig, URI, DATABASE, COLLECTION);
+        if (!result.isSuccess()) {
+            throw new PrepareFailException(getPluginName(), PluginType.SOURCE, result.getMsg());
+        }
+
+        this.params = ConfigBeanFactory.create(extractConfig, MongodbParameters.class);
+
+        if (extractConfig.hasPath(SCHEMA)) {
+            Config schema = extractConfig.getConfig(SCHEMA);
+            this.rowType = SeaTunnelSchema.buildWithConfig(schema).getSeaTunnelRowType();
+        } else {
+            this.rowType = SeaTunnelSchema.buildSimpleTextSchema();
+        }
+
+        // TODO: use format SPI
+        // default use json format
+        String format;
+        if (extractConfig.hasPath(FORMAT)) {
+            format = extractConfig.getString(FORMAT);
+            this.deserializationSchema = null;
+        } else {
+            format = DEFAULT_FORMAT;
+            this.deserializationSchema = new JsonDeserializationSchema(false, false, rowType);
+        }
+    }
+
+    @Override
+    public void setSeaTunnelContext(SeaTunnelContext seaTunnelContext) {
+        this.seaTunnelContext = seaTunnelContext;
+    }
+
+    @Override
+    public Boundedness getBoundedness() {
+        return JobMode.BATCH.equals(seaTunnelContext.getJobMode()) ? Boundedness.BOUNDED : Boundedness.UNBOUNDED;
```

Comment:
I think static data from mongo is `BOUNDED`, unless CDC.

Reply:
Makes sense, done